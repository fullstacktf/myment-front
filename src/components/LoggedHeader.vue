<template>
    <div class="headerNav">
        <img src="../assets/M.png" alt="our coolest logo">
        <newsBar class = "news"></newsBar>
    </div>
</template>

<script>
import newsBar from './NewsBar';
export default {
    components:{
        'newsBar': newsBar
    }

}
</script>

<style>
.news{
    height: 60px;
}
.headerNav{
    display:flex;
    border: 1px solid black;
}
.headerNav img {
    border: 1px solid black;
    width: 60px;
    height: 60px;
}

</style>
