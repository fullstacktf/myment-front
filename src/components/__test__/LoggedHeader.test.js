import LoggedHeader from '../LoggedHeader.vue';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import Buefy from 'buefy';
import mock from '../../test/mockBack';

describe('LoggedHeader', () => {
  let localVue;
  const mock = { newsLetter: "Hi, I'm a new bar" };
  beforeEach(() => {
    const Vue = createLocalVue();
    Vue.use(Buefy);
    localVue = Vue;
  });
  test('render component', () => {
    const wrapper = shallowMount(LoggedHeader, {
      localVue,
      data() {
        return { mock };
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('Check props', () => {
    const wrapper = shallowMount(LoggedHeader, {
      localVue,
      data() {
        return mock;
      },
    });
    expect(wrapper.find('newsbar-stub').attributes().newnews).toBe(
      mock.newsLetter,
    );
  });

  test('check Contents props', () => {});
});
