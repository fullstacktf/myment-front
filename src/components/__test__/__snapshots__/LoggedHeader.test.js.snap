// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoggedHeader render component 1`] = `"<div class=\\"headerNav\\"><img src=\\"../../public/assets/M.png\\" alt=\\"our coolest logo\\"> <newsbar-stub newnews=\\"\\" class=\\"news\\"></newsbar-stub></div>"`;
